toolkit/Makefile

```diff
@@ -59,6 +59,8 @@ REBUILD_DEP_CHAINS              ?= y
 HYDRATED_BUILD                  ?= n
 TARGET_ARCH                     ?=
 ALLOW_TOOLCHAIN_REBUILDS        ?= n
+RESOLVE_CYCLES_FROM_UPSTREAM    ?= n
```

Comment:
>RESOLVE_CYCLES_FROM_UPSTREAM [](http://example.com/codeflow?start=0&length=28)

Sorry to keep tormenting you over this, but I realized I haven't fully understood when we'd end up with a need to resolve cycles from upstream repos. I'm guessing we've hit that at some point during our ptests for delta builds, but if that's the case, I'm curious if that isn't due to a bug in the way we trick the toolkit into running delta builds.

Could you provide a real-life scenario, where we've hit such cycles? I'm curious what it was, to better understand the need for this change.  It feels like for all officially supported scenarios (and delta builds) we shouldn't need to break cycles with upstream packages.

Reply:
sure, we needed this feature when getting new packages to Mariner. This is not added to solve the bug in delta builds, but this PR will fix that bug. (Bug -> When we stopped building all RPMS in delta builds, build is getting blocked due to cycles. With this feature, only the blocking rpms will be fetched, and the build will be unblocked)

This is the scenario where we required cycle resolution from upstream.
We are trying to build a package called pandoc for a customer which has a major dependency on  ghc. 
When building the **ghc** package, we saw a lot of circular dependencies. Because the package is a compiler, it needed itself and some of the packages it provides, to build. This created multiple circular dependencies, which caused the build to fail at the grapher stage due to cycles. To resolve these cycles, we got the required ghc rpms locally and placed them in the out/RPMs folder. This unblocked the build locally.

To be able to build the ghc package in a pipeline, we'd need to use rpms available in upstream repos. So added this feature to the toolkit.

This is documented in a two-pager by @0xba1a . I'm not sure if I can share it here so I DM'ed it to you.